File: c#/one.cs
using System;
class Program {
    static int Days(int S, int N, int M) {
        if (M > N)
            return -1;
        int sundays = S / 7;
        int foodNeed = S * M;
        int shopWork = S - sundays;
        int minDays = (foodNeed + N - 1) / N;

        if (minDays > shopWork)
            return -1;
        return minDays;
    }
    static void Main() {
        Console.Write("Введите число S: ");
        int S = int.Parse(Console.ReadLine());
        Console.Write("Введите число N: ");
        int N = int.Parse(Console.ReadLine());
        Console.Write("Введите число M: ");
        int M = int.Parse(Console.ReadLine());
        int result = Days(S, N, M);
        if (result == -1)
            Console.WriteLine("Выжить невозможно");
        else
            Console.WriteLine($"Минимальное количество дней {result}");
    }
}

File: kotlin/one.kt
fun main() {
    print("Введите число S: ")
    val s = readLine()!!.toInt()
    print("Введите число N: ")
    val n = readLine()!!.toInt()
    print("Введите число M: ")
    val m = readLine()!!.toInt()
    val result = ostrov(s, n, m)
    if (result == -1) {
        println("Вжить невозможно")
    } else {
        println("Минимальное количество дней: $result")
    }
}
fun ostrov(s: Int, n: Int, m: Int): Int {
    if (m > n) {
        return -1
    }
    val foodNeed = s * m
    val sundays = s / 7
    val workDays = s - sundays
    val minDays = (foodNeed + n - 1) / n
    return if (minDays > workDays) -1 else minDays
}
